src/Chapter1.hs

```diff
@@ -536,8 +539,11 @@ value after "=" where the condition is true.
 
 Casual reminder about adding top-level type signatures for all functions :)
 -}
-
-mid x y z = error "mid: not implemented!"
+mid :: Ord a => a -> a -> a -> a
+mid x y z
+  | x <= y && y <= z || x >= y && y >= z = y
+  | y <= x && x <= z || y >= x && x >= z = x
+  | x <= z && z <= y || x >= z && z >= y = z
```

Comment:
As we mentioned in the course, the compiler in Haskell always need to ensure that the pattern-matching is exhaustive.
In here, you have a smart logic on comparisons and compiler can't figure out if it covers all cases or not.
To make it happy, you need to provide a final case, which would be something like: `| otherwise = ...`.

Reply:
Oh, yes it was showing a warning.